### PR TITLE
Add more nlinalg ops

### PR DESCRIPTION
**Pull request by twiecki**
Extends #50. There still is an issue with Ops that return more than one value.

### CODE REVIEW

File: tests/sandbox/test_jax.py [hunk 1]
```diff
@@ -23,6 +24,7 @@ def compare_jax_and_py(fgraph, inputs, cmp_fn=np.allclose):
     jax_res = theano_jax_fn(*inputs)
 
     if isinstance(jax_res, list):
+        # For some reason this returns [[array0, array1], None]
         assert all(isinstance(res, jax.interpreters.xla.DeviceArray) for res in jax_res)
```
Comment (twiecki) [inline tests/sandbox/test_jax.py:28]: Here is the current issue, not sure why this is happening.

File: theano/sandbox/jax_linker.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def create_jax_thunks(self, compute_map, storage_map):
 
             # JIT-compile the functions
             if len(node.outputs) > 1:
-                assert len(jax_funcs) == len(node.ouptputs)
+                assert len(jaxed_fgraph_outputs) == len(node.outputs)
```
Comment (twiecki) [inline theano/sandbox/jax_linker.py:79]: This had a typo before, probably need a test to cover this case.

Comment (brandonwillard) [reply theano/sandbox/jax_linker.py:79]: I don't believe the `jax_funcs` part is incorrect; there should be a JAXable function for each node output, which is what that assert is stating (aside from the typo).

**Comment by brandonwillard** [comment]
Wait, I'm confused; why don't we make these changes in #50?

File: tests/sandbox/test_jax.py (hunk 1 above)
Comment (brandonwillard) [reply tests/sandbox/test_jax.py:27]: The SVD `Op` is one of the rare ones that has two outputs.  That's why you're seeing this.

File: theano/sandbox/jax_linker.py (hunk 2 above)
Comment (twiecki) [reply theano/sandbox/jax_linker.py:79]: Hm, there was only one for multi-output func graphs.

**Comment by twiecki** [comment]
@brandonwillard Just because #50 should be ready to merge and this is still WIP.

**Comment by brandonwillard** [comment]
FYI: This PR tests some multi-output conditions that weren't previously covered, so it will need to contain those updates to move forward.

**Comment by brandonwillard** [comment]
All right, I closed the other PR and got everything working in this one.

**Comment by twiecki** [comment]
Thanks for the help @brandonwillard!

**Pull request merged**